xpk.py

```diff
@@ -1838,6 +1838,49 @@ def setup_docker_image(args) -> tuple[int, str]:
 
   return 0, docker_image
 
+def get_gke_outlier_dashboard(args):
+  """Get the identifier of GKE outlier dashboard deployed in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    str:
+      identifier of outlier dashbord if deployed in project,
+      None otherwise.
+  """
+  outlier_dashboard_filter = "displayName:'GKE - TPU Monitoring Dashboard'"
+  command = (
+      'gcloud monitoring dashboards list'
+      f' --project={args.project} --filter="{outlier_dashboard_filter}" --format="value(name)"'
+  )
+
+  return_code, return_value = run_command_for_value(command, 'GKE Dashboard List', args)
+
+  if return_code != 0:
+    xpk_print(f'GKE Dashboard List request returned ERROR {return_code}')
+    return None
+
+  if not return_value:
+    xpk_print(
+        f'No dashboard with {outlier_dashboard_filter} found in the'
+        f' project:{args.project}.'
+    )
+    xpk_print(
+        'Follow https://github.com/google/cloud-tpu-monitoring-debugging'
+        ' to deploy monitoring dashboard to view statistics and outlier mode of GKE metrics.'
+    )
+
+  dashboards = return_value.strip().split('\n')
+  if len(dashboards) > 1:
+    xpk_print(f'Multiple dashboards with same {outlier_dashboard_filter} exist in the project:{args.project}.')
```

Comment:
Interesting edge case, how would someone fix this?

Reply:
Would Cloud Monitoring actually let this happen? Or would they block it at creation time?